controllers/review.controller.js

```diff
@@ -0,0 +1,186 @@
+
+    review = await this.reviewLib.updateReview(id, rawData);
+    return res.status(202).json({
+      success: true,
+      data: review,
+    });
+  });
+
+  /**
+   * @desc Edit review to add a like
+   * @route PUT /api/v1/reviews/:id/likes
+   * @access Private
+   */
+  putReviewLike = asyncHandler(async (req, res, next) => {
+    // Add user to req.body
+    const { user, params, body } = req;
+    const { id } = params;
+    const { action } = body;
+
+    if (!['like', 'unlike'].includes(action)) {
+      return next(new ErrorResponse('Invalid action', 400));
+    }
+
+    let review = await this.reviewLib.fetchReview({ _id: id });
+    if (!review) {
+      return next(
+        new ErrorResponse(`Review with id: ${id} does not exist on the database`, 404),
+      );
+    }
+    let actionObj;
+    if (action === 'like') {
+      actionObj = {
+        $addToSet: { likes: ObjectID(user.id) },
+      };
+    }
+    if (action === 'unlike') {
+      actionObj = {
+        $pull: { likes: ObjectID(user.id) },
+      };
+    }
+
+    review = await this.reviewLib.updateReview(id, actionObj);
+    return res.status(202).json({
+      success: true,
+      data: review,
+    });
+  });
+
+  /**
+   * @desc Get reviews
+   * @route GET /api/v1/reviews
+   * @access Private
+   */
+  getReviews = asyncHandler(async (req, res) => {
+    const { query, params } = req;
+    const page = parseInt(query.page, 10);
+    const limit = parseInt(query.limit, 10);
+    let localQuery = { ...query };
+
+    if (params.bookId) {
+      localQuery = { ...localQuery, book: params.bookId };
+    }
+    const result = await advancedResults(Review, localQuery, { page, limit });
+
+    res.status(200).json({
+      success: true,
+      ...result,
+    });
+  });
+
+  /**
+   * @desc Get review
+   * @route GET /api/v1/reviews/:id
+   * @access Private
+   */
+  getReview = asyncHandler(async (req, res, next) => {
+    // Add user to req.body
+    const { id } = req.params;
+    req.body.authorInformation = req.user;
+
+    const review = await this.reviewLib.fetchReview({ _id: id });
+    if (!review) {
+      return next(
+        new ErrorResponse(`Review with id: ${id} does not exist on the database`, 404),
+      );
+    }
+    return res.status(200).json({
+      success: true,
+      data: review,
+    });
+  });
+
+  /**
+   * @desc Get review likes
+   * @route GET /api/v1/reviews/:id/likes
+   * @access Private
+   */
+  getReviewLikes = asyncHandler(async (req, res, next) => {
+    // Add user to req.body
+    const { id } = req.params;
+    req.body.authorInformation = req.user;
+
+    const review = await this.reviewLib.fetchReview({ _id: id }, { populate: 'likes', select: 'likes' });
+    if (!review) {
+      return next(
+        new ErrorResponse(`Review with id: ${id} does not exist on the database`, 404),
+      );
+    }
+    return res.status(200).json({
+      success: true,
+      data: review.toObject({ virtuals: true }),
+    });
```

Comment:
I was thinking you gonna return only the likes array without the review data so there is a difference between this API and the other one(GET /api/v1/reviews/:id). The way it is now, the other Review API can be modified a bit to serve this same purpose like `GET /api/v1/reviews/:id?includeLikes=true`